packages/monorepo-builder/tests/Git/MostRecentTagForCurrentBranchResolverTest.php

```diff
@@ -0,0 +1,56 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Symplify\MonorepoBuilder\Tests\Git;
+
+use PHPUnit\Framework\MockObject\MockObject;
+use PHPUnit\Framework\TestCase;
+use Symplify\MonorepoBuilder\Git\MostRecentTagForCurrentBranchResolver;
+use Symplify\MonorepoBuilder\Release\Process\ProcessRunner;
+use Symplify\MonorepoBuilder\Release\Process\ProcessRunnerInterface;
+
+final class MostRecentTagForCurrentBranchResolverTest extends TestCase
+{
+    private ProcessRunner|MockObject $processRunner;
+
+    protected function setUp(): void
+    {
+        $this->processRunner = $this->createMock(ProcessRunnerInterface::class);
+    }
+
+    public function testReturnMostRecentTagStringWithoutNewlineCharacters(): void
+    {
+        $this->processRunner
+            ->method('run')
+            ->with(MostRecentTagForCurrentBranchResolver::COMMAND, 'git-directory')
+            ->willReturn("\r1.0.0\r\n")
+        ;
```

Comment:
The mock is not needed here

Reply:
Why not? As we're testing `MostRecentTagForCurrentBranchResolver` and `ProcessRunner` is a dependency it seems ok for me to mock this method.